tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -1342,14 +1400,15 @@ def _train_multi(self, X, y, models: List[AbstractModel], hyperparameter_tune_kw
             n_repeats = self.n_repeats
         if (k_fold == 0) and (n_repeats != 1):
             raise ValueError(f'n_repeats must be 1 when k_fold is 0, values: ({n_repeats}, {k_fold})')
-        if time_limit is None:
+        feature_prune = feature_prune_kwargs is not None
+        if time_limit is None and not feature_prune:
```

Comment:
Remove `feature_prune` variable, simplify:

`if time_limit is None and feature_prune_kwargs is None`